### PR TITLE
fix(menu): avoid error when no facet values are retrieved

### PR DESCRIPTION
**Pull request by rayrutjes**
Works around a bug in the Menu component implementation:
image

cc @Shipow

### CODE REVIEW

File: src/components/Menu.vue [hunk 1]
```diff
@@ -51,7 +51,11 @@ export default {
         this.sortBy
       );
 
-      return data;
+      if (Array.isArray(data)) {
```
Comment (Haroenv) [inline src/components/Menu.vue]: in which case can this happen, since the default value is `[]`

**Comment by Haroenv** [comment]
What about adding `|| []` to https://github.com/algolia/vue-instantsearch/blob/db113a41c3f7796f7473634a55477a1269b03b1b/src/store.js#L311 instead?

File: src/components/Menu.vue (hunk 1 above)
Comment (rayrutjes) [reply src/components/Menu.vue]: This can happen when `getFacetValues` returns `{data: null}` like in the tests.

File: src/components/Menu.vue [hunk 2]
```diff
@@ -46,12 +46,16 @@ export default {
 
   computed: {
     facetValues() {
-      const { data = [] } = this.searchStore.getFacetValues(
```
Comment (rayrutjes) [inline src/components/Menu.vue]: @iam4x FYI: default value assignment when destructuring only kicks on `undefined` values.
The helper often returns `null` values, which is the case here so we can't solely rely on this.

**Pull request merged**